lib/service/awsoidc.go

```diff
@@ -0,0 +1,301 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package service
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/gravitational/trace"
+	"github.com/jonboulle/clockwork"
+	"github.com/sirupsen/logrus"
+
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/api/utils/retryutils"
+	"github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/teleport/lib/automaticupgrades"
+	"github.com/gravitational/teleport/lib/integrations/awsoidc"
+	"github.com/gravitational/teleport/lib/utils/interval"
+)
+
+const (
+	// updateDeployAgentsInterval specifies how frequently to check for available updates.
+	updateDeployAgentsInterval = time.Minute * 30
+)
+
+func (process *TeleportProcess) initDeployServiceUpdater() error {
+	// start process only after teleport process has started
+	if _, err := process.WaitForEvent(process.GracefulExitContext(), TeleportReadyEvent); err != nil {
+		return trace.Wrap(err)
+	}
+
+	resp, err := process.getInstanceClient().Ping(process.GracefulExitContext())
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	if !resp.ServerFeatures.AutomaticUpgrades {
+		return nil
+	}
+
+	process.log.Infof("The new service has started successfully. Checking for deploy service updates every %v.", updateDeployAgentsInterval)
+
+	// If criticalEndpoint or versionEndpoint are empty, the default stable/cloud endpoint will be used
+	var criticalEndpoint string
+	var versionEndpoint string
+	if automaticupgrades.GetChannel() != "" {
+		criticalEndpoint, err = url.JoinPath(automaticupgrades.GetChannel(), "critical")
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		versionEndpoint, err = url.JoinPath(automaticupgrades.GetChannel(), "version")
+		if err != nil {
+			return trace.Wrap(err)
+		}
+	}
+
+	issuer, err := awsoidc.IssuerFromPublicAddress(process.proxyPublicAddr().Addr)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	clusterNameConfig, err := process.getInstanceClient().GetClusterName()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	updater, err := NewDeployServiceUpdater(DeployServiceUpdaterConfig{
+		Log:                 process.log.WithField(trace.Component, teleport.Component(teleport.ComponentProxy, "deployserviceupdater")),
```

Comment:
```suggestion
		Log:                 process.log.WithField(trace.Component, teleport.Component(teleport.ComponentProxy, "awsoidc-deployserviceupdater")),
```
Or similar.

I really think we should mention the `aws-oidc`
Because deploy service are two very generic terms.